lib/haskell/natural4/src/LS/XPile/MathLang/MathLang.hs

```diff
@@ -381,11 +223,24 @@ gml2ml exp =
 
   EIs left right -> gml2ml exp {GML.exp = EVarSet left right}
 
-  _ ->
-    trace [i|\ngml2ml: not supported #{exp}\n|]
-      pure $ MathVar [i|gml2ml: not implemented yet #{expExp}|]
+  ENot body -> do
+    mlBody <- exp2pred body
+    pure $ MathPred (PredNot Nothing mlBody)
```

Comment:
This is an addition, it allows us to fully parse the vermin-and-rodents csv example into explainable MathLang.

Reply:
Cool, thanks! There was no reason to not have it before, simply that I hadn't thought of transpiling rodents and vermin into mathlang. 😁